examples/MQ/pixelDetector/src/PixelDigiSource.cxx

```diff
@@ -14,154 +14,147 @@
+    return atoi(tempstr.c_str());
 }
 
 Int_t PixelDigiSource::ReadEvent(UInt_t i)
 {
-  fDigis->Delete();
-  fNDigis = 0;
-
-  if (i == 0) {
-    fInputFile.clear();
-    fInputFile.seekg(0, std::ios::beg);
-  }
-
-  fCurrentEntryNo = i;
-
-  std::string buffer;
-  LOG(debug) << "PixelDigiSource::ReadEvent() Begin of (" << fDigis->GetEntries() << ")";
-  do {
-    getline(fInputFile,buffer);
-    LOG(debug) << "read from file: \"" << buffer << "\"";
-    if (buffer.find("EVENT BEGIN") == 0) {
-      fRunId     = ReadIntFromString(buffer,"RUNID");
-      fMCEntryNo = ReadIntFromString(buffer,"MCENTRYNO");
-      fPartNo    = ReadIntFromString(buffer,"PARTNO");
-      fEventHeader->SetRunId(fRunId);
-      fEventHeader->SetMCEntryNumber(fMCEntryNo);
-      fEventHeader->SetPartNo(fPartNo);
-
-      LOG(debug) << "GOT NEW EVENT " << fMCEntryNo << " (part " << fPartNo << ") with run id = " << fRunId;
+    fDigis->Delete();
+    fNDigis = 0;
+
+    if (i == 0) {
+        fInputFile.clear();
+        fInputFile.seekg(0, std::ios::beg);
     }
-    if (buffer.find("EVENT") == 0) continue;
-    Int_t detId     = atoi(buffer.c_str());
-    buffer.erase(0, buffer.find(' ') + 1);
-    Int_t feId      = atoi(buffer.c_str());
-    buffer.erase(0, buffer.find(' ') + 1);
-    Int_t col       = atoi(buffer.c_str());
-    buffer.erase(0, buffer.find(' ') + 1);
-    Int_t row       = atoi(buffer.c_str());
-    buffer.erase(0, buffer.find(' ') + 1);
-    Double_t charge = atof(buffer.c_str());
-    LOG(debug) << "    --/" << fNDigis << "/-->    " << detId << " / " << feId << " / " << col << " / " << row << " / " << charge;
-    new ((*fDigis)[fNDigis]) PixelDigi(-1,detId,feId,col,row,charge,0.);
-    fNDigis++;
-  } while (fInputFile && buffer.compare("EVENT END"));
-  LOG(debug) << "PixelDigiSource::ReadEvent() End of";
-
-  if (!fInputFile) {
-    return 1;
-  }
-
-  return 0;
-}
 
-Bool_t   PixelDigiSource::ActivateObject(TObject** obj, const char* BrName) {
-  if (strcmp(BrName,"PixelDigis"))
-    *obj = (TObject*)fDigis;
-  else if (strcmp(BrName,"EventHeader."))
-    *obj = (TObject*)fEventHeader;
-  else
-    return kFALSE;
+    fCurrentEntryNo = i;
+
+    std::string buffer;
+    LOG(debug) << "PixelDigiSource::ReadEvent() Begin of (" << fDigis->GetEntries() << ")";
+    do {
+        getline(fInputFile, buffer);
+        LOG(debug) << "read from file: \"" << buffer << "\"";
+        if (buffer.find("EVENT BEGIN") == 0) {
+            fRunId = ReadIntFromString(buffer, "RUNID");
+            fMCEntryNo = ReadIntFromString(buffer, "MCENTRYNO");
+            fPartNo = ReadIntFromString(buffer, "PARTNO");
+            fEventHeader->SetRunId(fRunId);
+            fEventHeader->SetMCEntryNumber(fMCEntryNo);
+            fEventHeader->SetPartNo(fPartNo);
+
+            LOG(debug) << "GOT NEW EVENT " << fMCEntryNo << " (part " << fPartNo << ") with run id = " << fRunId;
+        }
+        if (buffer.find("EVENT") == 0)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Inefficient usage of string::find() in condition; string::starts_with() could be faster.](https://app.codacy.com/manual/FairRootGroup/FairRoot/pullRequest?prid=5323596)